### PR TITLE
Introduce NanNewOneByteString

### PR DESCRIPTION
**Pull request by kkoopa**
This has been on the list for a long time. Don't really care much for it myself, I think it becomes ugly, but if we are to support lesser compilers, then so be it.

### CODE REVIEW

File: nan.h [hunk 1]
```diff
@@ -909,6 +924,32 @@ class NanEscapableScope {
...
+  }
+  }  // end of namespace imp
+  }  // end of namespace Nan
+
+  NAN_INLINE NanMaybeLocal<v8::String>
+  NanNewOneByteString(const uint8_t * value, int length = -1) {
+    std::vector<uint16_t> wideString;
+    Nan::imp::widenString(&wideString, value, length);
+    if (wideString.size() == 0) {
+      return Factory<v8::String>::return_t(v8::String::Empty());
```
Comment (bnoordhuis) [inline nan.h]: Just a note: you don't have to return String::Empty() yourself, V8 already does that.

File: nan.h [hunk 2]
```diff
@@ -909,6 +924,32 @@ class NanEscapableScope {
     return function_template->HasInstance(value);
   }
 
+  namespace Nan { namespace imp {
+  NAN_INLINE void
+  widenString(std::vector<uint16_t> *ws, const uint8_t *s, int l = -1) {
```
Comment (bnoordhuis) [inline nan.h]: Does `l` need a default value?

**Comment by bnoordhuis** [comment]
LGTM

**Comment by kkoopa** [comment]
Fixed. CI is not working currently because of 2600bd138d3cd1c1c01414ca663d44a2ebd9d053 and the fact that the new tap version has not been published to npm (it contains a showstopper bug fix for windows and the tag is in the git repository). I won't downgrade tap again, will just wait for the new version to get published.

**Pull request merged**